### DISCUSSION TITLE
Client Idle timeout implementation

### DISCUSSION THREAD

**Discussion by Geo-7** [tokio-rs/tokio#4768] [Q&A]
I have  created a simple proxy with tokio ( [example](https://github.com/tokio-rs/tokio/blob/master/examples/proxy.rs) )
 is there any builtin function which could help me to have a feature like HAPROXY client timeout?
I do not want to interrupt active connections I want to close idle connections , using tokio::time::timeout on future did not help because It will close active connections.

**Comment by Darksonn** [comment]
You can put timeouts on individual operations on the socket with `tokio::time::timeout`.

**Reply by Geo-7** [reply]
That's not gonna help, I want to start timing from last socket activity, as I told I don't  want to close active connection.

**Reply by sfackler** [reply]
Wrap the socket in something that resets timers on socket activity.

**Reply by Geo-7** [reply]
@sfackler thanks for the suggestion, As I understand there is no builtin function for this. I'll try to find a way to do that.
